src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,76 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import React from 'react';
+import { Filter, FilterEnum } from '../../types/Filter';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type Props = {
+  filter: Filter;
+  onFilter: (f: Filter) => void;
+};
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+export const TodoFilter: React.FC<Props> = ({ filter, onFilter }) => {
+  const handleChangeSelect = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    onFilter({
+      ...filter,
+      select: event.target.value as FilterEnum,
+    });
+  };
+
+  const handleChangeInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    onFilter({
+      ...filter,
+      input: event.target.value,
+    });
+  };
+
+  const handleClearInput = () => {
+    onFilter({
+      ...filter,
+      input: '',
+    });
+  };
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            data-cy="statusSelect"
+            value={filter.select}
+            onChange={handleChangeSelect}
+          >
+            {Object.values(FilterEnum).map(value => (
+              <option value={value} key={value}>
+                {value.at(0)?.toUpperCase() + value.slice(1)}
+              </option>
+            ))}
```

Comment:
Here it is now impossible to understand what is happening, transfer the calculations to separate variables and helpers and use ready-made results with good naming.